### DISCUSSION TITLE
how changesets determine which packages changed

### DISCUSSION THREAD

**Discussion by tjx666** [changesets/changesets#1088] [General]
I change a file of one package, and commit it. When I run `changesets`, there shows no changed packages.

![image](https://user-images.githubusercontent.com/41773861/218078379-dff7b054-8331-4f6a-be1b-9f84d0ea1c35.png)

**Comment by Andarist** [comment]
It compares the changed packages in your current HEAD against the state in the `baseBranch` (usually the `main` or the `master` branch). It just checks if any files changed within a given package directory - but this can be also customized with [`changedFilePatterns`](https://github.com/changesets/changesets/pull/1033) setting

**Reply by buquan** [reply]
hello, i config `master` and change `testc.js` and `commit` . after that excute `pnpm changeset`  but i cant get changed packages as follow images ,why? 

`@changesets/cli @2.26.0`

<img width="1431" alt="image" src="https://user-images.githubusercontent.com/23711051/218618981-1f68ee39-de82-40f7-8fe1-b5cc92bafd14.png">

<img width="1427" alt="image" src="https://user-images.githubusercontent.com/23711051/218619032-73bb4983-26e1-4638-870c-70f553fa6e01.png">

**Reply by Andarist** [reply]
It would be best if you could share a repro case for this. It's hard to debug this using screenshots.

**Reply by buquan** [reply]
> It would be best if you could share a repro case for this. It's hard to debug this using screenshots.

https://github.com/buquan/mySets
thanks a lot

**Reply by Andarist** [reply]
On the screenshots - I don't see any "pending changes" in the left sidebar. Did you actually change any files? It would be great to know your exact repro steps and the time at which you expected to see changed packages.

I tested the provided repo and it detects `testc` as changed in both of those scenarios
```sh
git checkout -b test-things-out && echo >> packages/testc/index.js && pnpm changeset
```
```sh
git checkout -b test-things-out && touch packages/testc/other.js && git add packages/testc/other.js && pnpm changeset
```

What I noticed though is that this doesn't end up being detected as changed:
```sh
git checkout -b test-things-out && touch packages/testc/other.js && pnpm changeset
```
It's because this creates a new file and that file is not *yet* tracked by git.

**Reply by tjx666** [reply]
any third party package can detect the changed packages? @Andarist

**Reply by Andarist** [reply]
Could you rephrase? I'm not sure what you are asking about here.

**Reply by tjx666** [reply]
I'm write some node script, I need a library can detect the changed packages names
